### ISSUE TITLE
BatchNormalization center and scale flags swapped

### ISSUE THREAD

**Issue by EelcoHoogendoorn** [keras-team/keras#11488]
Please make sure that the boxes below are checked before you submit your issue.
If your issue is an **implementation question**, please ask your question on [StackOverflow](http://stackoverflow.com/questions/tagged/keras) or [on the Keras Slack channel](https://keras-slack-autojoin.herokuapp.com/) instead of opening a GitHub issue.

Thank you!

- [ ] Check that you are up-to-date with the master branch of Keras. You can update with:
`pip install git+git://github.com/keras-team/keras.git --upgrade --no-deps`

- [ ] Check that your version of TensorFlow is up-to-date. The installation instructions can be found [here](https://www.tensorflow.org/get_started/os_setup).

- [ ] Provide a link to a GitHub Gist of a Python script that can reproduce your issue (or just copy the script here if it is short).

I know the description above urges the use of latest versions, but I cant use those sadly; but I doubt it has any bearing on this bug anyway. These are the packages I am using:
TF 1.8.0, conda-forge
Keras 2.2.4, conda-forge

I have an input tensor with strictly positive values, that I would like to have normalised in scale; without losing the positivity.

From reading the docstring and underlying code, it seems clear that this should be accomplished with the center=False kwarg. However, in practice it seem that using scale=False gives me the expected high level behavior.

Digging in a little deeper, I can indeed verify that I pass on only positive values; yet that right after the BN with center=False, I may end up with negative values (even though they ought only to have been scaled). And this does not happen with scale=False.

It seems to me that somewhere in the (very long) calling chain, these flags are getting swapped in meaning. I havnt been able to pin down where this happens; but as for a stylistic suggestion, that may be related to this... There is a lot of calling of functions with lots of arguments going on, without using kwarg names explicitly at the calling site. Having a long list of positional arguments without an obvious implied ordering makes for code that is very hard to check for correctness!

**Comment by EelcoHoogendoorn** [comment]
fwiw; upgrading to TF 1.11 does not change a thing

**Comment by EelcoHoogendoorn** [comment]
`
import numpy as np

import keras


input_layer = keras.layers.Input(shape=(1,), name='input')

normalized_input_layer = keras.layers.BatchNormalization(center=False, name='normalized')(input_layer)

output_layer = keras.layers.Dense(1, name='output')(normalized_input_layer)

model = keras.models.Model(
    inputs=[input_layer],
    outputs=[output_layer, normalized_input_layer],
)

input = np.random.uniform(0, 100, size=10000)

output = input / 100

model.compile(
    optimizer='adam',
    loss={'output': 'mean_squared_logarithmic_error'},
)

model.fit(
    input,
    output,
    batch_size=100,
    epochs=5,
)

prediction, normalized_input = model.predict(input)

assert input.min() > 0
assert normalized_input.min() > 0
`

Luckily it reproduces with quite a minimal example; however my minimal example is even worse; for some reason I now lose my non-negativity regardless of my choice of flags.

**Comment by gabrieldemarmiesse** [comment]
By default, batch normalization does this operation:

`output = (x - mean) / sqrt(var + epsilon) * gamma + beta`

By setting `center=False`, you remove the beta from the equation. The equation becomes

`output = (x - mean) / sqrt(var + epsilon) * gamma`

As you can see we still substract the mean, so it's logical for outputs values to be positive and negatives. There is no problem here.

**Comment by EelcoHoogendoorn** [comment]
Ah ok I assumed the toggling of the two would be coupled, considering the existence one is motivated by the other. Could have gotten that from reading to docstring more closely I suppose... but disable_beta/disable_gamma would also be more self-descriptive names..

Is there a canonical way in keras to get the behavior I am after?

**Comment by gabrieldemarmiesse** [comment]
maybe something like `Lambda(lambda x: x / K.mean(x, axis=-1, keepdims=True))`? I'm not sure exatly, but it doesn't seem difficult to do if you use a lambda layer and the keras backend.

**Comment by EelcoHoogendoorn** [comment]
Dunno... I think the moving mean with momentum is pretty important for my application. Batches are fairly small, outliers are huge, and quantitative precision is important.

**Comment by gabrieldemarmiesse** [comment]
I think your best bet is to rewrite the batchnorm layer then. There is no easy option for that in keras.
